### ISSUE TITLE
No /30 net in SSL VPN since 21.7

### ISSUE THREAD

**Issue by RudiRausF** [opnsense/core#5212]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Since OPNsense 21.7 all site to site SSL VPN tunnel stop working.
it is not possible to ad a VPN site to site tunnel with a /30 tunnel net.
existing tunnels stop working.
same configuration on client site is working

**To Reproduce**

1. Go to 'VPN/OpenVPN/Server/'
2. Click on 'add server or existing server'
3. Scroll down to 'save'
4. See error (The following input errors were detected:
A prefix longer than 28 cannot be used with a net30 topology.)

**Expected behavior**

Tunnel net with 2 host addresses. (first server second client)

**Environment**

OPNsense 21.7.2_1-amd64FreeBSD 12.1-RELEASE-p20-HBSDOpenSSL 1.1.1l 24 Aug 2021
CPU type | AMD EPYC Processor (4 cores)

**Comment by iperry-indigex** [comment]
This was introduced back in July due to an OpenVPN pool size configuration change referenced [here](https://github.com/OpenVPN/openvpn/pull/153). The relevant opnSense forum I was able to find [here](https://forum.opnsense.org/index.php?topic=23934.msg114092#msg114092). The problem with the both of these is that it appears that opnSense is attempting to use net30 by default. The error that comes up when creating a new VPN interface is "A prefix longer than 28 cannot be used with a net30 topology." The issue with this is even when creating a p2p vpn (which, I would think, should be using p2p topology rather than net30 or subnet), it forces the use of either net30 or a subnet topology if the topology checkbox is checked. As above, the OpenVPN devs see no valid use for a net30 or subnet topology VPN with less than 2 usable pool IPs. In the case of a p2p topology VPN, this occurs to me to be perfectly fine.

Long story short, prefix checking was implemented [here](https://github.com/opnsense/core/pull/5108/files/634f9334f81b4f9215bc3c5724e7d1408abb4202#diff-2c99e78eba540456ab3e81d6c2552fbcd6829bd71195210d0c2c9ad8f307afa4) but as far as I can see, it does not take into account peer-to-peer servers and treats all servers as net30 or subnet.

**Comment by kulikov-a** [comment]
sorry, it looks like I missed it. will make a pr tomorrow

**Comment by fichtner** [comment]
# opnsense-patch 0ee3ecde
